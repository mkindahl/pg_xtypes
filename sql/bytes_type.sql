SELECT '1 bytes'::bytes, '1 B'::bytes;
SELECT '1 KiB'::bytes, '1 kB'::bytes;
SELECT '1.2 KiB'::bytes, '1.2 kB'::bytes;
SELECT '2345 MiB'::bytes;
SELECT '1 GiB'::bytes;
SELECT '1 TiB'::bytes;
SELECT '1 EiB'::bytes;

create table sample(time timestamptz, size1b int8, size1 bytes, size2b int8, size2 bytes);

select setseed(1.0);

insert into sample
with vals as (
  select time, random() * 1024 as one, random() * 1024 * 1024 as two
    from generate_series('2022-01-01 00:00:00'::timestamptz,
    	 	         '2022-01-02 00:00:00'::timestamptz,
			 '1 minute') as time)
select time,
       one, format('%s MiB', one)::bytes, two, format('%s kB', two)::bytes
from vals;
     
\x on
select time,
       size1, pg_size_pretty(1024 * 1024 * size1b),
       size2, pg_size_pretty(1024 * size2b),
       size1 + size2 as sum,
       size1 - size2 as difference,
       size1 / 2 as divided,
       size1 * 2 as product1,
       2 * size1 as product2
  from sample
 where size1 > size2
 limit 5;
\x off
