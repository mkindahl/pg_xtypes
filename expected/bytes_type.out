SELECT '1 bytes'::bytes, '1 B'::bytes;
 bytes  | bytes  
--------+--------
 1.00 B | 1.00 B
(1 row)

SELECT '1 KiB'::bytes, '1 kB'::bytes;
  bytes  |  bytes  
---------+---------
 1.00 kB | 1.00 kB
(1 row)

SELECT '1.2 KiB'::bytes, '1.2 kB'::bytes;
  bytes  |  bytes  
---------+---------
 1.20 kB | 1.20 kB
(1 row)

SELECT '2345 MiB'::bytes;
  bytes  
---------
 2.29 GB
(1 row)

SELECT '1 GiB'::bytes;
  bytes  
---------
 1.00 GB
(1 row)

SELECT '1 TiB'::bytes;
  bytes  
---------
 1.00 TB
(1 row)

SELECT '1 EiB'::bytes;
  bytes  
---------
 1.00 EB
(1 row)

create table sample(time timestamptz, size1b int8, size1 bytes, size2b int8, size2 bytes);
select setseed(1.0);
 setseed 
---------
 
(1 row)

insert into sample
with vals as (
  select time, random() * 1024 as one, random() * 1024 * 1024 as two
    from generate_series('2022-01-01 00:00:00'::timestamptz,
    	 	         '2022-01-02 00:00:00'::timestamptz,
			 '1 minute') as time)
select time,
       one, format('%s MiB', one)::bytes, two, format('%s kB', two)::bytes
from vals;
     
\x on
select time,
       size1, pg_size_pretty(1024 * 1024 * size1b),
       size2, pg_size_pretty(1024 * size2b),
       size1 + size2 as sum,
       size1 - size2 as difference,
       size1 / 2 as divided,
       size1 * 2 as product1,
       2 * size1 as product2
  from sample
 where size1 > size2
 limit 5;
-[ RECORD 1 ]--+-----------------------------
time           | Sat Jan 01 00:02:00 2022 PST
size1          | 565.71 MB
pg_size_pretty | 566 MB
size2          | 539.73 MB
pg_size_pretty | 540 MB
sum            | 1.08 GB
difference     | 25.98 MB
divided        | 282.86 MB
product1       | 1.10 GB
product2       | 1.10 GB
-[ RECORD 2 ]--+-----------------------------
time           | Sat Jan 01 00:06:00 2022 PST
size1          | 770.04 MB
pg_size_pretty | 770 MB
size2          | 677.96 MB
pg_size_pretty | 678 MB
sum            | 1.41 GB
difference     | 92.09 MB
divided        | 385.02 MB
product1       | 1.50 GB
product2       | 1.50 GB
-[ RECORD 3 ]--+-----------------------------
time           | Sat Jan 01 00:10:00 2022 PST
size1          | 295.93 MB
pg_size_pretty | 296 MB
size2          | 159.51 MB
pg_size_pretty | 160 MB
sum            | 455.44 MB
difference     | 136.42 MB
divided        | 147.96 MB
product1       | 591.86 MB
product2       | 591.86 MB
-[ RECORD 4 ]--+-----------------------------
time           | Sat Jan 01 00:13:00 2022 PST
size1          | 211.86 MB
pg_size_pretty | 212 MB
size2          | 104.55 MB
pg_size_pretty | 105 MB
sum            | 316.41 MB
difference     | 107.31 MB
divided        | 105.93 MB
product1       | 423.72 MB
product2       | 423.72 MB
-[ RECORD 5 ]--+-----------------------------
time           | Sat Jan 01 00:15:00 2022 PST
size1          | 917.07 MB
pg_size_pretty | 917 MB
size2          | 141.56 MB
pg_size_pretty | 142 MB
sum            | 1.03 GB
difference     | 775.51 MB
divided        | 458.54 MB
product1       | 1.79 GB
product2       | 1.79 GB

\x off
